libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -30,6 +30,8 @@ import (
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
+var AllSupportedHints = []string{"enabled", "module", "metricsets", "host", "period", "timeout", "metricspath", "username", "password", "stream", "processors", "multiline", "json", "disable", "ssl", "metrics_filters", "raw", "include_lines", "exclude_lines", "fileset", "pipeline"}
```

Comment:
does `metricspath` actually works? shouldn't it be `metrics_path` based on https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-autodiscover-hints.html#_co_elastic_metricsmetrics_path ?

where `disable` is coming from?

Also as I understand it is a merged list for the metricbeat and for the filebeat, correct? can you please add a comment for that? Is there any way to split it?

Reply:
I changed metricspath to metrics_path ! Nice catch, thanks Tania!

Disable comes from this: https://github.com/elastic/elastic-agent-autodiscover/blob/7490558445cb8d3248462255edc8ac64bc18b8eb/utils/hints_test.go#L177
We can disable logs collection apparently

I have added a comment in [7d2c429](https://github.com/elastic/beats/pull/38213/commits/7d2c4298be62f28c2003cb617d0dbfec0568427f)

Currently we parse annotations based on the fact that start with the  prefix ([L209](https://github.com/elastic/elastic-agent-autodiscover/blob/7490558445cb8d3248462255edc8ac64bc18b8eb/utils/hints.go#L209 )) and then we split with "/" symbol ([L215](https://github.com/elastic/elastic-agent-autodiscover/blob/7490558445cb8d3248462255edc8ac64bc18b8eb/utils/hints.go#L215))
So currently this is a unified solution for elastic-agent and beats. I deliberately had moved the solution to the elastic-agent-autodiscovery in oder not to introduce  an additional annotation loop and different implementations. So splitting it for beats will require additional checks if we have Prefix.logs or Prefix.metrics. I consider it out of scope here, mainly because the main reason behind of those changes it to offer the feature in elastic-agent and not in beats. Hope it makes sense